circuit_knitting/cutting/qpd/qpd.py

```diff
@@ -315,11 +318,24 @@ def _(gate: RXXGate | RYYGate | RZZGate | CRXGate | CRYGate | CRZGate):
     return QPDBasis(maps, coeffs)
 
 
-@_register_qpdbasis_from_gate("csx")
-def _(gate: CSXGate):
-    retval = qpdbasis_from_gate(CRXGate(np.pi / 2))
-    for operations in unique_by_id(m[0] for m in retval.maps):
-        operations.insert(0, TGate())
+@_register_qpdbasis_from_gate("csx", "cs", "csdg", "cp")
+def _(gate: CSXGate | CSGate | CSdgGate | CPhaseGate):
+    theta = _theta_from_gate(gate)
+    if gate.name == "csx":
+        retval = qpdbasis_from_gate(CRXGate(theta))
+    else:
+        retval = qpdbasis_from_gate(CRZGate(theta))
+    if gate.name in {"csx", "cs"}:
+        for operations in unique_by_id(m[0] for m in retval.maps):
+            operations.insert(0, TGate())
+    elif gate.name in {"csdg"}:
+        for operations in unique_by_id(m[0] for m in retval.maps):
+            operations.insert(0, TdgGate())
+    else:
+        assert gate.name in {"cp"}
+        for operations in unique_by_id(m[0] for m in retval.maps):
+            operations.insert(0, PhaseGate(theta / 2))
```

Comment:
This would have fewer branches and I actually think this would be easier to reason about if it were at least two distinct methods.  `cs`, `csdg`, and `cp` might naturally all go together since they are controlled rotations about the Z axis, but I think if will be more clear if `csx` remains its own method, at the very least.

Reply:
> This would have fewer branches and I actually think this would be easier to reason about if it were at least two distinct methods. `cs`, `csdg`, and `cp` might naturally all go together since they are controlled rotations about the Z axis, but I think if will be more clear if `csx` remains its own method, at the very least.

Agree, I think breaking this out into 3 functions looks pretty natural.